nats/aio/client.py

```diff
@@ -707,6 +712,14 @@ def _drain_sub(self, sid):
                 nonlocal sub
                 nonlocal err_cb
 
+                semaphore = asyncio.Semaphore(sub.max_cb_concurrency)
```

Comment:
I think creating the semaphore object should be optional for when `sub.max_cb_concurrency` is greater than 1, wdyt?

Reply:
maybe the semaphore might be a `sub` property? I personally don't like accessing variables declared in conditional blocks from other places. 
`if sub.is_async:` would have to become `if sub.is_async and sub.max_cb_concurrency > 1:` 

TBH I would take this chance to refactor this part of code a bit more. Maybe we can get rid of `is_async` and demand that `sub.cb` is a coroutine (remove `coro` parameter). I am aware that it will need few more refactors (and probably a braking change), but it might be a right time for it. 
This way current  code:

```
                            # Invoke depending of type of handler.
                            if sub.coro is not None:
                                if sub.is_async:
                                    # NOTE: Deprecate this usage in a next release,
                                    # the handler implementation ought to decide
                                    # the concurrency level at which the messages
                                    # should be processed.
                                    self._loop.create_task(sub.coro(msg))
                                    await semaphore.acquire()
                                    self._loop.create_task(
                                        release_semaphore_after_cb(msg)
                                    )
                                else:
                                    await sub.coro(msg)
                            elif sub.cb is not None:
                                if sub.is_async:
                                    raise NatsError(
                                        "nats: must use coroutine for async subscriptions"
                                    )
                                else:
                                    # Schedule regular callbacks to be processed sequentially.
                                    self._loop.call_soon(sub.cb, msg)
```
Will become simpler:
```
                            if sub.max_cb_concurrency > 1:
                                await semaphore.acquire()
                                self._loop.create_task(
                                    release_semaphore_after_cb(msg)
                                )
                            else:
                                await sub.cb(msg)
```

Reply:
I like those ideas a lot 👍  I'm ok as well with a breaking change if it makes things simpler as well.

Reply:
Cool, I'll see what I can do 🙂